src/gateway/WebSocket.ts

```diff
@@ -45,73 +58,72 @@ export default function Socket(token: string, wsURL: string, intents: number) {
     }));
   }
 
-  function resume() {
-    if (ws.readyState === WebSocket.OPEN) {
-      ws.send(JSON.stringify({
-        op: GATEWAY_OPCODES.RESUME,
-        d: {
-          token,
-          session_id,
-          seq,
-        },
-      }));
-    }
+  newWS() {
+    this.ws = new WebSocket(this._wsURL);
+    this.ws.on('open', this.identify);
+    this.ws.on('close', this.onClose);
+    this.ws.on('message', this.onMessage);
+  }
+
+  onClose(code: number) {
+    if (!GATEWAY_CLOSE_EVENT_CODES[code]) console.log(code);
+    else console.log(GATEWAY_CLOSE_EVENT_CODES[code]);
   }
 
-  function onMessage(data: Payload) {
+  onMessage(data: Payload) {
     data = JSON.parse(String(data));
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const { t, s, op, d } = data;
-    if (s) seq = s;
+    if (s) this.seq = s;
 
     switch (op) {
       case GATEWAY_OPCODES.DISPATCH:
-      // if (EVENTS[t]) client.emit(EVENTS[t], d);
-        session_id = d.session_id;
+        // if (EVENTS[t]) client.emit(EVENTS[t], d);
+        this.session_id = d.session_id;
         break;
 
       case GATEWAY_OPCODES.HEARTBEAT:
-        heartbeat(false);
+        this.heartbeat(false);
         break;
 
       case GATEWAY_OPCODES.RECONNECT:
-        clearInterval(interval);
-        ws.removeEventListener('close');
-        ws.terminate();
+        clearInterval(this.interval);
+        this.ws.removeEventListener('close');
```

Comment:
I checked ws code, and it extends directly EventEmitter. You should use `off` or `removeListener` instead of `removeEventListener`. I would probably use `off` since you use `on` and not `addListener`.